profile.install

```diff
@@ -86,9 +88,147 @@ function profile_uninstall() {
   }
 }
 
+
+/**
+ * Implements hook_requirements().
+ *
+ * This is here to deal with an update numbering problem in early versions of
+ * this module (updates started numbering at 10000, not 1000). If the wrong
+ * number is detected for the last update version, we fix it so that updates
+ * proceed properly going forward.
+ */
+function profile_requirements($phase) {
+  if (backdrop_get_installed_schema_version('profile') == 10000) {
```

Comment:
It's probably better to only have this check during the `update` and `install` phases. If that still works for this. Otherwise it'll always be running this query on every page load.

see https://api.backdropcms.org/api/backdrop/core%21modules%21system%21system.api.php/function/hook_requirements/1

Reply:
This is difficult. We'd had [a Zulip conversation about this issue](https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/hook_update_N.20numbering). The problem is that an earlier version gave the wrong update number (10000, not 1000), which means that any further properly numbered updates wouldn't get called (1001 being less than 10000), and use of numbers >7000 are not allowed anyhow. The issue isn't relevant for fresh installations.

[Laryn suggested using hook_requirements](https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/hook_update_N.20numbering/near/214141433), so I'd adopted that approach.

It looks like hook_requirements() is not called on every page load, but only during `install`, `update`, and called with `runtime` during display of the status page. Is that incorrect?

Reply:
Yeah, I had seen that conversation. The idea might be the best option. I figured it might be possible to reduce the times when it checks the schema version by limiting it to only being during the install or update phases.

Just to be safe, you may also want to ensure that you return what the hook expects. So if you've got no actual status message to return, then do `return NULL` like here https://api.backdropcms.org/api/backdrop/core%21includes%21install.core.inc/function/install_verify_requirements/1.

Reply:
Even though NULL appears OK as a return value, the docs do say to return an array, so I've added an empty array as the return value.

Also removed the check during 'install' phase (since it's not needed at first installation).

And, upon further investigation, doing the check-and-fix during 'update' phase happens early enough (before checking update numbering) that the numbering will get fixed during the routine update.php that people are supposed to always run after upgrading a module.

If a person forgets to run update.php, the admin bar or status page normally warns them that there are unapplied updates; but with the mis-numbering, this won't happen with this module. However, the first time they check the status page, the 'runtime' phase will be called, which will fix the update numbering, and thereafter they'll get nagged to run update.php.

And last, it appears that hook_requirements() is not called on every page load, so there's no each-page penalty.

So, I think the current version now fully works!